### PR TITLE
Object to primitive conversion

### PR DESCRIPTION
**Pull request by mahdiHashemi14**
Please review and comment/commit if anything is wrong

### CODE REVIEW

File: 1-js/04-object-basics/09-object-toprimitive/article.md [hunk 1]
```diff
@@ -120,32 +120,32 @@ let user = {
...
-If there's no `Symbol.toPrimitive` then JavaScript tries to find methods `toString` and `valueOf`:
+اگر `Symbol.toPrimitive` وجود نداشته باشد، سپس جاوااسکریپت سعی می‌کند که متدهای `toString` و `valueOf`  را پیدا کند:
 
-- For the "string" hint: `toString`, and if it doesn't exist, then `valueOf` (so `toString` has the priority for string conversions).
-- For other hints: `valueOf`, and if it doesn't exist, then `toString` (so `valueOf` has the priority for maths).
+- برای جزء "string": `toString` و اگر این متد وجود نداشت، سپس `valueOf` (پس `toString` برای تبدیل‌های رشته‌ای اولویت دارد).
+- برای بقیه جزءها: `valueOf` و اگر این متد وجود نداشت، سپس `toString` (پس `valueOf` برای ریاضیات اولویت دارد).
 
-Methods `toString` and `valueOf` come from ancient times. They are not symbols (symbols did not exist that long ago), but rather "regular" string-named methods. They provide an alternative "old-style" way to implement the conversion.
+متدهای `toString` و `valueOf` از زمان‌های گذشته وجود دارند. آن‌ها سمبل نیستند (سمبل‌ها انقدر قدیمی نیستند) بلکه متدهای «معمولی» هستند که اسمی رشته‌ای دارد. آن‌ها راهی جایگزین که «سبک قدیمی» دارد را برای پیاده‌سازی تبدیل فراهم می‌کنند.
```
Comment (mahdiHashemi14) [inline 1-js/04-object-basics/09-object-toprimitive/article.md:139]: Line 139 needs more attention.

File: 1-js/04-object-basics/09-object-toprimitive/article.md [hunk 2]
```diff
@@ -250,28 +250,28 @@ let obj = {
   }
 };
 
-alert(obj + 2); // 22 ("2" + 2), conversion to primitive returned a string => concatenation
+alert(obj + 2); // تبدیل به مقدار اصلی، یک رشته => ادغام برگرداند، (2 + "2") 22
```
Comment (mahdiHashemi14) [inline 1-js/04-object-basics/09-object-toprimitive/article.md:253]: More attention, please!

File: 1-js/04-object-basics/09-object-toprimitive/article.md [hunk 3]
```diff
@@ -250,28 +250,28 @@ let obj = {
...
+2. در غیر این صورت اگر جزء `"string"` بود
+    - متدهای `obj.toString()` و `obj.valueOf()` را امتحان کن، هر کدام که وجود داشت.
+3. در غیر این صورت اگر جزء `"number"` یا `"default"` بود
+    - متدهای `obj.valueOf()` و `obj.toString()` را امتحان کن، هر کدام که وجود داشت.
 
-In practice, it's often enough to implement only `obj.toString()` as a "catch-all" method for string conversions that should return a "human-readable" representation of an object, for logging or debugging purposes.  
+در عمل، اینکه فقط `obj.toString()` را به عنوان متدی «همه‌گیر» برای تبدیل‌های رشته‌ای که باید یک نمایش «قابل خواندن برای انسان» از شیء را برگدانند، برای اهداف دیباگ یا لاگ گرفتن، اغلب اوقات کافی است.
 
-As for math operations, JavaScript doesn't provide a way to "override" them using methods, so real life projects rarely use them on objects.
+همینطور برای عملیات ریاضی، جاوااسکریپت راهی برای باطل کردن آن‌ها با استفاده از متدها را فراهم نمی‌کند، پس پروژه‌های زندگی واقعی به ندرت از آن‌ها روی شیءها استفاده می‌کنند.
```
Comment (mahdiHashemi14) [inline 1-js/04-object-basics/09-object-toprimitive/article.md:277]: These two lines (275 and 277) need more attention, too.

File: 1-js/04-object-basics/09-object-toprimitive/article.md (hunk 1 above)
Comment (mahdyar) [reply 1-js/04-object-basics/09-object-toprimitive/article.md:139]:
```suggestion
متدهای `toString` و `valueOf` از زمان‌های گذشته وجود دارند. آن‌ها سمبل نیستند (سمبل‌ها انقدر قدیمی نیستند) بلکه متدهای «معمولی» هستند که اسمی رشته‌ای دارد. آن‌ها راهی جایگزین برای پیاده‌سازی تبدیل به «سبک قدیمی» را  فراهم می‌کنند.
```

File: 1-js/04-object-basics/09-object-toprimitive/article.md (hunk 2 above)
Comment (mahdyar) [reply 1-js/04-object-basics/09-object-toprimitive/article.md:253]:
```suggestion
alert(obj + 2); // تبدیل به مقدار اصلی یک رشته برگرداند => ادغام، (2 + "2") 22
```

**Review by mahdyar — APPROVED** [review]
Please review the suggestion(s).

**Review by mahdyar — CHANGES_REQUESTED** [review]
Please review the suggestion(s).

**Comment by mahdiHashemi14** [comment]
Thank you @mahdyar 
I also had a question: do you think "جزء" is a good translation for "hint"? For example in [line 35](https://github.com/javascript-tutorial/fa.javascript.info/blob/2a31b9f2d7c657562fbf560bb6a5355a690c41c6/1-js/04-object-basics/09-object-toprimitive/article.md?plain=1#L35).

/done

**Review by mahdyar — APPROVED** [review]
> Thank you @mahdyar 
I also have a question: do you think "جزء" is a good translation for "hint"? For example in [line 35](https://github.com/javascript-tutorial/fa.javascript.info/blob/2a31b9f2d7c657562fbf560bb6a5355a690c41c6/1-js/04-object-basics/09-object-toprimitive/article.md?plain=1#L35).
/done 

Yes, that LGTM.
Thank you, @mahdiHashemi14!

**Pull request merged**